crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -129,41 +133,87 @@ pub fn execute_call_entry_point(
+                let error_trace = context.error_trace();
                 EntryPointExecutionError::VirtualMachineExecutionErrorWithTrace {
-                    trace: context.error_trace(),
-                    source: error,
+                    trace: error_trace[..min(10000, error_trace.len())].to_string(),
+                    source: internal_error,
                 }
             }
-            other_error => other_error,
+            other_error => {
+                context
+                    .error_stack
+                    .push((storage_address, format!("{}\n", &other_error)));
+                other_error
+            }
         }
     });
 
-    let identifier = match entry_point.call_type {
-        CallType::Call => AddressOrClassHash::ContractAddress(entry_point.storage_address),
-        CallType::Delegate => AddressOrClassHash::ClassHash(entry_point.class_hash.unwrap()),
-    };
-    runtime_state.cheatnet_state.trace_data.exit_nested_call(
+    // region: Modified blockifier code
+    map_and_process_entry_point_call_result(
+        entry_point_call_result,
+        runtime_state,
         resources,
-        CallResult::from_execution_result(&result, &identifier),
-    );
+        entry_point,
+        context,
+    )
+    // endregion
+}
 
+fn map_and_process_entry_point_call_result(
+    result: EntryPointExecutionResult<(CallInfo, SyscallCounter)>,
+    runtime_state: &mut RuntimeState,
+    resources: &mut ExecutionResources,
+    entry_point: &mut CallEntryPoint,
+    context: &mut EntryPointExecutionContext,
+) -> EntryPointExecutionResult<CallInfo> {
     result
-    // region: Modified blockifier code
-    // We skip recursion depth decrease
-    // endregion
+        .map_err({
+            |error| {
+                runtime_state.cheatnet_state.trace_data.exit_nested_call(
+                    resources,
+                    &Default::default(),
+                    CallResult::from_err(
+                        &error,
+                        &AddressOrClassHash::get_by_entry_point(entry_point),
+                    ),
+                );
+                error
+            }
+        })
+        .map(|(call_info, this_call_syscall_counter)| {
+            let versioned_constants = context.tx_context.block_context.versioned_constants();
+            // We don't want the syscall resources to pollute the results
+            *resources -= &versioned_constants
+                .get_additional_os_syscall_resources(&this_call_syscall_counter)
+                .expect("Could not get additional resources");
+            let nested_syscall_counter_sum = aggregate_nested_syscall_counters(
+                &runtime_state
+                    .cheatnet_state
+                    .trace_data
+                    .current_call_stack
+                    .top(),
+            );
+            let syscall_counter =
+                sum_syscall_counters(&nested_syscall_counter_sum, &this_call_syscall_counter);
```

Comment:
You can do it way easier by saving the syscall counter state before the execution and doing the subtraction of syscall counters. You can reuse the existing abstraction for that - just add syscall counter to `CallStackElement` (it was there before in the resources, but as you know the blockifier structs changed). 

If we also agree on the idea I proposed above too it will let us unify the execution branches (error and success handling) and move it back to the original function